worker/filenotifywatcher/watcher.go

```diff
@@ -165,40 +165,39 @@ func (w *Watcher) loop() error {
 				continue
 			}
 			// If the event is not a create or delete event, ignore it.
-			if maskType(event.Mask) == unknown {
+			opType := parseOpType(event.Op)
+			if opType == unknown {
 				continue
 			}
 
-			created := event.Mask&inotify.InCreate != 0
-
 			if w.logger.IsTraceEnabled() {
 				w.logger.Tracef("dispatch event for fileName %q: %v", w.fileName, event)
 			}
 
-			w.changes <- created
+			w.changes <- opType == created
```

Comment:
https://github.com/juju/juju/pull/16416/files#diff-4f0eb9dc333bf74abdb56f3beaf2fe784faf085eeac582660f5d1105a14d9749R196-R203

We only care about created or deleted. We parse that into a type (created, deleted or unknown) which then we filter on, so we know we either created or deleted. That way we can just say if it was created or removed.

Reply:
ok, so from a logic point of view, it would have been the same to write:
```go
w.changes <- true
```
or
```go
w.changes <- struct{}
```
am I right?

Reply:
no

See: https://github.com/juju/juju/blob/main/internal/changestream/stream/stream.go#L224-L239